tests/undefined_symbol_in_sdfg_test.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+
+import pytest
+import numpy as np
+
+import dace
+from dace import symbolic
+
+
+def test_undefined_symbol_in_sdfg():
+    """Tests that an UndefinedSymbol can be used in an SDFG but raises an error when compiled."""
+    
+    @dace.program
+    def program_with_undefined_symbol(A: dace.float64[20, 20], B: dace.float64[20]):
+        # Create a transient with an undefined dimension
+        undefined_dim = symbolic.UndefinedSymbol()
+        tmp = dace.define_local([20, undefined_dim], dace.float64)
```

Comment:
If `tmp`'s dimensions were swapped, i.e., the shape was `[undefined_dim, 20]` and the access was `tmp[0, i]`, would code generation work without error?